### PR TITLE
Refactor code to untie platform specific code dependency on millis calculation draft

### PR DESCRIPTION
**Pull request by boshtannik**
Refactor code to untie platform specific code dependency on millis calculation

### CODE REVIEW

File: src/mesh_lib/mod.rs [hunk 1]
```diff
@@ -31,18 +28,20 @@ pub struct NodeConfig {
     // between `Node`s
     pub usart: Usart,
 
-    // The hardware abstraction library timer, which will be used for
-    // counting of milliseconds.
-    pub millis_timer: TC0,
+    // A closure for calculating millis
+    pub millis_fn_ptr: fn() -> ms,
```
Comment (berkus) [inline src/mesh_lib/mod.rs]: Maybe better:

```
trait PlatformTime {
  fn get_millis() -> ms;
}
```

mm?

Comment (boshtannik) [reply src/mesh_lib/mod.rs]: Maybe. and how it shall be organised within the struct? get_millis: impl PlatformTime ?

Comment (boshtannik) [reply src/mesh_lib/mod.rs]: Do not know, how to do it easier, and more flexible. Throw this PlatformTime trait implemented structure by the reference trough all the project structures, where they are needed to be used, or make static reserved memory allocated place, and save initialized timer there, to let use the timer from statically allocated place.
2023-12-11-043605_1324x641_scrot

Comment (boshtannik) [reply src/mesh_lib/mod.rs]: @berkus, I have made some updates to use PlatformTime trait, instead function pointer. Made it as mothod, not as static associated function. Look please trough, tell what could be odd in this code, and could be simplified?

Comment (boshtannik) [reply src/mesh_lib/mod.rs]: Not actual anymore

Comment (boshtannik) [reply src/mesh_lib/mod.rs]: @berkus . Long story short. I'am merging with no review?

**Pull request merged**